apluslms-yamlidator/apluslms_yamlidator/document.py

```diff
@@ -192,7 +193,9 @@ def load(cls, path, version=None, **kwargs):
         try:
             document = container.get_latest(max_version=version)
         except KeyError:
-            document = cls(container, None, Dict(), version)
+            document = cls(container, None, Dict(version=str(cls.version)), version)
```

Comment:
I presume there should be `str(version)` instead of `str(cls.version)`.

Reply:
This should also handle the case when version is None. In addition, it should use `version_key`. This probably should also support creating empty documents with initialization function, as the schema might require some values (e.g. steps in the project config).

So maybe something:

```
except KeyError:
  if version is not None:
    document = cls(container, None, Dict(), version)
    documnet.initialize_data()
    document.validate() # should validate as long as initialize_data ensures the initial date is there
 ```

and in Document:

```
def initialize_data(self):
  vkey = self.version_key
  if vkey:
    missing = vkey not in self._data
    self._data[vkey] = str(self.version)
    #if missing and hasattr(self._data, 'move_to_end'):
    #  self._data.move_to_end(vkey, last=False)
```

Reply:
what should be done if version_key is empty?

Reply:
what should be in initialize? all data required for passing validation is version

Reply:
If version key is empty, then the schema probably doesn't require it. In that case, it could even violate the schema.

For RomanConfig/ProjectConfig we could add the default or the legacy configuration. 50/50 if that is a good idea, though. And there is no need for subclass to overwrite the mehtod if they don't need to.

Reply:
so... add defaults to version in the schemas?

Reply:
Version is a bit special child and it's probably better to handle this way and bit specially.